### PR TITLE
Feature for #1078: Configurable swaylock colors

### PR DESCRIPTION
**Pull request by 4e554c4c**
Colors are configured through the command line so that swaylock conforms
to the i3lock fork 'github.com/chrjguill/i3lock-color'. Differences from
it are that one letter options '-r' and '-s' are not implimentend because
'-s' is already used by '--scaling' in swaylock.
This commit also fixed whitespace in 'include/swaylock/swaylock.h' and
changed `parse_color` in 'common/util.h' so that it can accept colors
that do not start with a hash. This was done to keep compatability with
the i3lock fork.

### CODE REVIEW

File: swaylock/main.c [hunk 1]
```diff
@@ -431,6 +480,59 @@ int main(int argc, char **argv) {
 				exit(EXIT_FAILURE);
 			}
 			break;
+		case 0:
+			if (strcmp(long_options[option_index].name, "font") == 0) {
+				free(config->font);
+				config->font = strdup(optarg);
+			}
+			else if (strcmp(long_options[option_index].name, "line-uses-ring") == 0) {
```
Comment (SirCmpwn) [inline swaylock/main.c]: style nit: join the } and else if lines together

File: swaylock/main.c [hunk 2]
```diff
@@ -431,6 +480,59 @@ int main(int argc, char **argv) {
...
+					exit(EXIT_FAILURE);
+				}
+				line_source = 1;
+			}
+			else if (strcmp(long_options[option_index].name, "line-uses-inside") == 0) {
+				if (line_source != 0) {
+					sway_log(L_ERROR, "Line source options conflict");
+					exit(EXIT_FAILURE);
+				}
+				line_source = 2;
```
Comment (SirCmpwn) [inline swaylock/main.c]: Can you make an enum for these?

File: swaylock/main.c [hunk 3]
```diff
@@ -664,23 +768,56 @@ void render(struct render_data *render_data) {
...
-				cairo_set_source_rgb(window->cairo, 0, 0, 0);
+				cairo_set_source_u32(window->cairo, config->colors.separator);
 				cairo_arc(window->cairo, wwidth/2, wheight/2, ARC_RADIUS, highlight_start, highlight_start + TYPE_INDICATOR_BORDER_THICKNESS);
 				cairo_stroke(window->cairo);
 
 				cairo_arc(window->cairo, wwidth/2, wheight/2, ARC_RADIUS, highlight_start + TYPE_INDICATOR_RANGE, (highlight_start + TYPE_INDICATOR_RANGE) + TYPE_INDICATOR_BORDER_THICKNESS);
 				cairo_stroke(window->cairo);
 			}
 
+			if (line_source == 1) {
```
Comment (SirCmpwn) [inline swaylock/main.c]: Use a switch here please

File: swaylock/main.c [hunk 4]
```diff
@@ -664,23 +768,56 @@ void render(struct render_data *render_data) {
...
+				switch(render_data->auth_state) {
+				case AUTH_STATE_VALIDATING: {
+					cairo_set_source_u32(window->cairo, config->colors.validating.outer_ring);
+					break;
+				}
```
Comment (SirCmpwn) [inline swaylock/main.c]: These braces aren't necessary.

File: swaylock/swaylock.1.txt [hunk 5]
```diff
@@ -45,6 +45,43 @@ Options
...
+
+*--separatorcolor* <rrggbb[aa]>::
+	Sets the color of the lines that seperate highlight segments.
+
+*--keyhlcolor* <rrggbb[aa]>::
+	Sets the color of keypress highlight segments.
+
+*--bshlcolor* <rrggbb[aa]>::
+	Sets the color of backspace highlight segments.
+
```
Comment (SirCmpwn) [inline swaylock/swaylock.1.txt]: Seems docs are missing for line-uses-ring? Possibly others?

Comment (SirCmpwn) [inline swaylock/swaylock.1.txt]: Can you alphabetize the options and put all the appearance-related ones under a new section called Appearance?

**Comment by SirCmpwn** [comment]
It might be best to break swaylock compatability and change how the scaling mode is specified so we can gain compatability with this existing tool, like we have compatability with i3-gaps. Does anyone have an opinion on this? Also, -r doesn't conflict with any existing options, why not implement it?

File: swaylock/main.c (hunk 2 above)
Comment (4e554c4c) [inline swaylock/main.c]: Yes, I was considering it already.

**Comment by 4e554c4c** [comment]
Thanks for looking over this. Since -r and -s are for related features I elected to implement both or neither so I'll go and do both. Also the new usage string seems crowded to me, should I change it?

**Comment by SirCmpwn** [comment]
Yeah, you can probably just leave all of the appearance options out of the usage string. Maybe add a "for more details see man swaylock" or something.

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**